components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -1164,16 +1164,17 @@ - (void)setTrackingScrollView:(UIScrollView *)trackingScrollView {
 #endif  // #if 0
 #endif  // #if DEBUG
 
-  // If this header is shared by many scroll views then we leave the insets when switching the
-  // tracking scroll view.
-  if (!_sharedWithManyScrollViews) {
-    [self fhv_removeInsetsFromScrollView:_trackingScrollView];
-  }
+  UIScrollView *oldTrackingScrollView = _trackingScrollView;
 
   BOOL wasTrackingScrollView = _trackingScrollView != nil;
-
   _trackingScrollView = trackingScrollView;
 
+  // If this header is shared by many scroll views then we leave the insets when switching the
+  // tracking scroll view.
+  if (!_sharedWithManyScrollViews && wasTrackingScrollView) {
+    [self fhv_removeInsetsFromScrollView:oldTrackingScrollView];
```

Comment:
should this method guard against removing it from the existing _trackingScrollView?